deploy/stacks/backend_stack.py

```diff
@@ -362,13 +362,16 @@ def create_opensearch_serverless_stack(self):
         )
 
     def get_s3_prefix_list(self):
-        ec2_client = boto3.client("ec2", region_name=self.region)
-        response = ec2_client.describe_prefix_lists(
-            Filters=[
-                {
-                    'Name': 'prefix-list-name',
-                    'Values': [f'com.amazonaws.{self.region}.s3']
-                },
-            ]
-        )
+        try:
+            ec2_client = boto3.client("ec2", region_name=self.region)
+            response = ec2_client.describe_prefix_lists(
+                Filters=[
+                    {
+                        'Name': 'prefix-list-name',
+                        'Values': [f'com.amazonaws.{self.region}.s3']
+                    },
+                ]
+            )
+        except:
+            return ""
```

Comment:
I am not particularly happy with this piece. I need to review other alternatives

Reply:
We could read from environment variables the GITHUB_ACTIONS env var. The issue is that this stack is alter executed in the CICD pipelines. I think we would need to add the env var to the CodeBuild stage, a bit complex for just this.

The main issue is that the except is returning a valid output instead of trowing an exception. In a real deployment, it could happen that CodeBuild has no permissions to list S3 prefixes, then the prefix list id is set to "". Later when we set the security groups for ECS and CodeBuild we add outbound rules to that list id. It will be giving permissions to a wrong id. My guess is that the CloudFormation will fail as there are no prefix ids as empty strings. 

```
# Add S3 Prefix List Connection
            if s3_prefix_list:
                sg_connection.allow_to(
                    ec2.Connections(peer=ec2.Peer.prefix_list(s3_prefix_list)),
                    ec2.Port.tcp(443),
                    'Allow ECS Task to S3 Prefix List'
                )
```